src/cljs/athens/electron.cljs

```diff
@@ -325,6 +325,48 @@
   ;; open or create a new starter db
 
   ;; Watch filesystem, e.g. in case db is updated via Dropbox sync
+  
+    ;; Update index.transit    
+  (defn write-db-index [file filepath]
+    (.. fs (writeFile filepath file (fn [e] (dispatch [:boot/desktop]))))
+  )
+  
+  ;; Create new index.transit db
+  (defn create-db-index [file filepath]
+    (.. fs (writeFile filepath file (fn [e] (dispatch [:fs/open-dialog]))))
+  ) 
+  
+  (defn open-dialog-index
+    "Allow user to open a Backup file."
+    [filepath]
+    (js/alert "Your index.transit is corrupted, please open up a backup")
+    (let [res  (.showOpenDialogSync dialog (clj->js {:properties ["openFile"]
+                                                     :filters    [{:name "Transit" :extensions ["transit.bkp"]}]}))
+          open-file (first res)]
+      (when (and open-file (.existsSync fs open-file))
+        (let [read-db (.readFileSync fs open-file)
+              db-file      (try  (dt/read-transit-str read-db)(catch  :default e ((js/console.error (js/Error. e))
+                                                                                  (open-dialog-index filepath))))              
+              ] 
+           (if (= (:schema db-file) db/schema) 
+                  ((write-db-index read-db filepath))(open-dialog-index filepath))))
+           (let [confirm (js/window.confirm (str "No file selected. Would you like to create a new db?"))]
+                              (if confirm 
+                              (create-db-index "[]" filepath)
+                              (open-dialog-index filepath))))
+  ) 
+
+  ;; Handle index.transit 
+  (defn handle-index-transit [filepath]  
+    (let [read-db (.readFileSync fs filepath)
+          db-file    (try  (dt/read-transit-str read-db)(catch  :default e ((js/console.error (js/Error. e))
```

Comment:
Same error here, invoking result of `js/console.error` which is `nil` as `fn`.

Reply:
@neotyk  i am not sure about the result here since i try and call two functions one for logging and then open the dialog box
thanks

Reply:
you should then wrap it in `do`

Reply:
done 
thanks